File: Java i Cpp/tydzien_3/zadanie_1/WierszTrojkataPascala.java
public class WierszTrojkataPascala {
    int liczby[];

    int Silnia(int n) {
        if(n == 0 || n ==1) {
            return 1;
        }
        return n * Silnia(n - 1);
    }

    public WierszTrojkataPascala (int n) {
        liczby = new int[n + 1];
        int chwilowe = Silnia(n);
        for(int indeks = 0; indeks <= n; indeks +=1) {
            liczby[indeks] = chwilowe / (Silnia(indeks) * Silnia(n - indeks));
        }
    }

    public int Liczba(int indeks) {
        return liczby[indeks];
    }
}

File: Java-Macyna/tydzien_3/zadanie_2/WierszTrojkataPascala.cpp
// WierszTrojkataPascala.cpp
#include <iostream>
#include "nagl.hpp"

WierszTrojkataPascala::WierszTrojkataPascala(int n) {
    liczby = new int[n + 1];
    int licznik = Silnia(n);

    for(int i = 0; i <= n; i += 1) {
        liczby[i] = licznik/(Silnia(i) * Silnia(n - i));
    }
}

WierszTrojkataPascala::~WierszTrojkataPascala() {
    delete[]liczby;
}

int WierszTrojkataPascala::Silnia(int n) {
    int wynik = 1;

    for(int i = 1; i <= n; i += 1) {
        wynik *= i;
    }

    return wynik;
}

int WierszTrojkataPascala::Liczba(int n) {
    return liczby[n];
}